data/address/address.d.ts

```diff
@@ -0,0 +1,17 @@
+import { OptionsOrAll } from "@data/generators"
+
+export type Address = {
+    address1: string,
+    address2?: string,
+    address3?: string,
+    address4?: string,
+    address5?: string,
+    postCode: string
+}
+
+export type AddressOptions = {
+    include: OptionsOrAll<AddressIncludes>
+}
+
+// If updating, consider the includes set of @util/address/addressGenerator
+export type AddressIncludes = 'address2' | 'city' | 'region' | 'country'
```

Comment:
Thinking of splitting the required parts of the Address from the optional. It'll make a little more work it makes Address a composite type but it means this could simply be run on "keyof" and so it couldn't deviate from those being requested.
Possible follow up to experiment with